Todo_app/src/components/useFetch.js

```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+import { useState, useEffect } from "react";
+import configService from "./config.js";
+
+export default function useFetch() {
+  const [tasks, setTasks] = useState([]);
+
+  useEffect(async () => {
+    const res = await axios.get(configService.todo_api);
+    let dbAllTasks = res.data;
+    setTasks(dbAllTasks);
+  }, []);
+
+  const getAllTasks = async () => {
+    const res = await axios.get(configService.todo_api);
+    let dbAllTasks = res.data;
+    setTasks(dbAllTasks);
+    return dbAllTasks;
+  };
+
+  const changeTasksToShow = async () => {
+    let dbAllTasks = await getAllTasks();
+    setTasks(dbAllTasks);
+  };
+
+  const hideAllDoneTasks = async () => {
```

Comment:
every time the filters change, ask from the server for a new tasks list (in the request, u should send filters & sortBy params